nvtabular/io.py

```diff
@@ -473,14 +480,28 @@ def _write_thread(self):
                 self.queue.task_done()
 
     def _write_metadata(self):
+        if self.cats is None:
+            return
         metadata_writer = open(os.path.join(self.out_dir, "metadata.json"), "w")
         data = {}
         data["file_stats"] = []
         for i in range(len(self.data_files)):
-            data["file_stats"].append({"file_name": f"{i}.data", "num_rows": self.num_samples[i]})
-        data["cats_name"] = self.cats
-        data["conts_name"] = self.conts
-        data["conts_labels"] = self.labels
+            data["file_stats"].append(
+                {"file_name": f"{i}.parquet", "num_rows": self.num_samples[i]}
+            )
+        # cats
+        data["cats"] = []
+        for c in self.cats:
+            data["cats"].append((c, self.col_idx[c]))
```

Comment:
Why do we need the col_idx data? Can't the reader just figure this out from either the parquet metadata or from the ordering of the cats/conts/labels columns?

Reply:
They need it on the hugectr side when reading the data, it looks like they cannot do it and they need the index.